### PR TITLE
Feature/evolve Evolve Asset

### PR DESCRIPTION
**Pull request by kudlaj**

### CODE REVIEW

File: src/services/EvolvingService.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      minterPvk: process.env.MINTER_PVK || '',
+      rpcMinter: process.env.RPC_MINTER || '',
+      minterLaosCollection: process.env.MINTER_LAOS_COLLECTION || '',
+    };
+    console.log("Evolving config: ", evolveConfig);
+    this.serviceHelper = new ServiceHelper(evolveConfig);
+  }
+
+  public async evolve(input: EvolveInput): Promise<EvolveResponse> {
+    const { tokenId, name, description, properties, image } = input;
```
Comment (davidpla) [inline src/services/EvolvingService.ts:20]: rename properties to attributes

File: src/services/EvolvingService.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    if (!image || 
+        (!image.startsWith("ipfs://") && 
+         !image.startsWith("data:image/") && 
+         !image.startsWith("https://ipfs.io/ipfs/"))) {
+      throw new Error("Invalid image format");
+    }
+
+    const attributes = this.serviceHelper.parseAssetAttributes(properties || '[]'); // Ensure attributes is an array
```
Comment (davidpla) [inline src/services/EvolvingService.ts:29]: rename to attributesParsed

File: src/resolvers/EvolveResolver.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { EvolveInput } from "../types/graphql/inputs/EvolveInput";
+import { EvolveResponse } from "../types/graphql/outputs/EvolveOutput";
+
+@Resolver()
+export class EvolveResolver {
+  constructor(private evolvingService: EvolvingService) {}
+
+  @Query(() => String)
+  evolveStatus() {
+    return "Evolution in progress!";
```
Comment (davidpla) [inline src/resolvers/EvolveResolver.ts:13]: change message to something like "use mutation"

File: src/services/MintingService.ts [hunk 4]
```diff
@@ -59,13 +26,13 @@ export class MintingService {
       throw new Error("Invalid image format");
     }
 
-    const attributes = this.parseAssetAttributes(properties || '[]'); // Ensure attributes is an array
+    const attributes = this.serviceHelper.parseAssetAttributes(properties || '[]');
```
Comment (davidpla) [inline src/services/MintingService.ts:29]: rename properties to attributes

File: src/services/blockchain/LaosService.ts [hunk 5]
```diff
@@ -30,7 +30,6 @@ export class MintService {
   public async mintSingleNFT(params: MintSingleNFTParams): Promise<MintResult> {
```
Comment (davidpla) [inline src/services/blockchain/LaosService.ts:30]: rename to mint

File: src/services/blockchain/LaosService.ts [hunk 6]
```diff
@@ -75,9 +75,59 @@ export class MintService {
...
+            error.message
+          );
+          throw error;
+        });
+
+      const receipt = await this.retryOperation(
+        () => this.provider.waitForTransaction(tx.hash, 1, 14000),
+        20
+      );
+      const tokenId = this.extractTokenId(receipt, this.contract, 'EvolvedWithExternalURI');
```
Comment (davidpla) [inline src/services/blockchain/LaosService.ts:111]: can be removed?

File: src/resolvers/EvolveResolver.ts (hunk 3 above)
Comment (kudlaj) [reply src/resolvers/EvolveResolver.ts]: removed

**Pull request merged**